### PR TITLE
[api] Keep logits in fp32

### PR DESCRIPTION
**Pull request by stephenroller**
**Patch Description**
This patch keeps our logit scores in fp32 so they are more accurately reported to the user. Has the nice added benefit of fixing a known bug where using temperature 0.1 causes overflow issues.

**Testing steps**
@tbmihailov running evals on this

### CODE REVIEW

File: metaseq/sequence_generator.py [hunk 1]
```diff
@@ -249,8 +250,9 @@ def _generate(
                 tokens[:, : step + 1],
                 incremental_state=incremental_states,
             )
-            model_out[0].div_(self.temperature)
-            lprobs = self.model.get_normalized_probs(model_out, log_probs=True)
+            model_predictions = model_out[0].float()
```
Comment (suchenzang) [inline metaseq/sequence_generator.py:253]: Add a comment here for which bug this fixes? Don't want future eager optimizers to get excited about taking this out...

**Review by suchenzang — APPROVED** [review]

File: metaseq/sequence_scorer.py [hunk 2]
```diff
@@ -72,7 +72,7 @@ def gather_target_probs(probs, target):
             for bd, tgt, is_single in batched:
                 sample["target"] = tgt
                 curr_prob = model.get_normalized_probs(
-                    bd, log_probs=len(models) == 1
+                    bd[0], log_probs=len(models) == 1
```
Comment (lilisierrayu) [inline metaseq/sequence_scorer.py:75]: why not pass  bd[0].float() to be consistent with generator?

File: metaseq/sequence_generator.py (hunk 1 above)
Comment (stephenroller) [reply metaseq/sequence_generator.py:253]: we're currently in a live debugging session lmao
